fdbclient/NativeAPI.actor.cpp

```diff
@@ -2227,6 +2227,7 @@ void DatabaseContext::updateProxies() {
 	proxiesLastChange = clientInfo->get().id;
 	commitProxies.clear();
 	grvProxies.clear();
+	ssVersionVectorCache.clear();
```

Comment:
It appears "updateProxies()" gets called whenever "basicLoadBalance()" is called, and "basicLoadBalance()" is getting called at a lot of places in NativeAPI.actor.cpp. Is it overly aggressive to clear the version vector cache here? It is correct, but it may make the client pull the entire version vector more frequently. Or, is it the best we can do about finding out when to clear the cache?

Reply:
I don't think this is cleared too frequently. Notice at line 2225, we check if the `clientInfo->get().id` has changed, i.e., master changes or recovery. So the clear happens only after a recovery.

Reply:
Ok, good.

Reply:
I am still not sure why the assert in "VersionVector::getDelta()" didn't fail after recovery. (Also, this issue could cause wrong results even when PROXY_USE_RESOLVER_PRIVATE_MUTATIONS is disabled, not sure why we are not seeing more failures because of this.) I will try to investigate this further.